app-native/src/midi/coremidi.rs

```diff
@@ -0,0 +1,47 @@
+use super::{MidiDriver, MidiOutput};
+use hero_studio_core::time::{clock, ClockTime};
+
+#[derive(Clone)]
+pub struct CoreMidi {
+  name: String,
+}
+
+impl CoreMidi {
+  pub fn new() -> CoreMidi {
+    CoreMidi {
+      name: "Core MIDI".to_string(),
+    }
+  }
+}
+
+impl MidiDriver for CoreMidi {
+  fn get_host_time(&self) -> ClockTime {
+    let host_time = unsafe { external::AudioGetCurrentHostTime() };
+    let nanos = unsafe { external::AudioConvertHostTimeToNanos(host_time) };
+    ClockTime::new(
+      ((nanos as u128 * clock::NANOS_PER_SECOND as u128) / clock::UNITS_PER_SECOND as u128) as u64,
```

Comment:
Why this casting to 128 to cast it again to 64?

Reply:
I don't like this code very much, but to avoid a multiplication overflow I need to use a `u128` (2^64 * 2^64 = 2^(64 + 64) = 2^128). Afterwards I do a division which decrements the number of required bits back to 64. I could have done this operation using `f64` with minimal precision loss, but preferred to keep it as an integer at the expense of using a `u128`.